File: src/project.cpp
#include "project.h"

namespace orq
{
	// Define data in this scope
	DataContext *Project::data;

	Project::Project(QString path)
	{
		// Append .orq if needed
		if (!path.endsWith(".orq"))
			path += ".orq";

		// Create database
		data = new DataContext(path);
	}

	Project::~Project()
	{
		// Close database when destroying object
		delete data;
		data = nullptr;
		// Close database connections
		DataContext::close();
	}

	QHash<int, QPair<ItemType, int>> Project::getVersions()
	{
		qFatal("error: Project::getVersions() is not implemented");
	}

	DataContext& Project::getData()
	{
		return *data;
	}
}

File: openrq/project.py
# from PySide2.QtCore import QString, QHash, QPair

iDataContext = __import__("datacontext")

class Project:
	def __init__(self, path):
		self.open = True
		if path.endswith(".orq"):
			path += ".orq"
		self.data = iDataContext.DataContext(path)
		return
	def getVersions():
		pass

	def getData():
		pass
				